pkg/tfbridge/schema.go

```diff
@@ -1652,6 +1652,32 @@ func isDefaultOrZeroValue(tfs shim.Schema, ps *SchemaInfo, v resource.PropertyVa
 func extractSchemaInputs(state resource.PropertyValue, tfs shim.Schema, ps *SchemaInfo,
 	rawNames bool) resource.PropertyValue {
 
+	if ps == nil {
+		ps = &SchemaInfo{}
+	}
+
+	for IsMaxItemsOne(tfs, ps) {
```

Comment:
It's fantastic to get quick fix out! IN the longer term maybe worth looking into abstracting out the traversal using path-based or some such methods, so the codebase deals with these idiosyncrasies in one place and gets it right once and for all.